trilium_alchemy/core/note/note.py

```diff
@@ -603,7 +602,7 @@ class MyNote(Note):
     ```
     """
 
-    note_id: str | None = EntityIdDescriptor()
+    note_id: EntityIdDescriptor | str | None = EntityIdDescriptor()

```

Comment:
Since this type is published in the documentation, I think we should keep this as `str | None` even if not technically correct in the case of accessing the class attribute `Note.note_id` (which would be unusual to do anyway). I think having a `# type: ignore` here is fine.